src/postprocessing/annotate-links.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Post-processing module that annotates links extracts with the spec
+   shortname they link to
+ */
+const fs = require('fs');
+const path = require('path');
+
+function canonicalizeUrl(url, options) {
+    options = options || {};
+
+    let canon = url.replace(/^http:/, 'https:')
+            .split('#')[0]
+            .replace('index.html', '')
+            .replace('Overview.html', '')
+            .replace('cover.html', '')
+            .replace(/spec.whatwg.org\/.*/, 'spec.whatwg.org/')  // subpage to main document in whatwg
+            .replace(/w3.org\/TR\/(([^\/]+\/)+)[^\/]+\.[^\/]+$/, 'w3.org/TR/$1') // subpage to main document in w3c
+            .replace(/w3.org\/TR\/([^\/]+)$/, 'w3.org/TR/$1/') // enforce trailing slash
+            .replace(/w3c.github.io\/([^\/]+)$/, 'w3c.github.io/$1/') // enforce trailing slash for ED on GitHub
+        ;
+
+    if (options.datedToLatest) {
+        canon = canon.replace(
+            /w3.org\/TR\/[0-9]{4}\/[A-Z]+-(.*)-[0-9]{8}\/?/,
+            'w3.org/TR/$1/');
+    }
+
+    let equivalentUrls = (options.equivalents) ? options.equivalents[canon] : null;
+    if (Array.isArray(equivalentUrls)) {
+        return (options.returnAlternatives ? equivalentUrls : equivalentUrls[0]);
+    }
+    else {
+        return (equivalentUrls ? equivalentUrls : canon);
+    }
+}
+
+const needsSaving = {};
+
+module.exports = {
+  dependsOn: ['links'],
+  input: 'crawl',
```

Comment:
Ah, the post-processor is a bit clunky here. This post-processing module can run directly once the spec has crawled so `input: 'spec'` would be the right choice in theory (this would also allow to use the default `save` logic that the post-processor contains instead of re-defining it because the post-processor does something else for crawl-wide post-processing modules).

That said, to be able to do that, you would need a way to access the list of specs in the crawl, which could either be done by updating the signature of the run function to `run: async function(spec, specs, options)` or by copying `speclist` into the options as you had done previously (no need to worry about memory copies a priori).

Anyway, no big deal for now. We can revisit that later on.